internal/core/src/index/InvertedIndexTantivy.cpp

```diff
@@ -447,6 +331,176 @@ InvertedIndexTantivy<T>::BuildWithRawData(size_t n,
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<double>(
+                    static_cast<const double*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::VarChar: {
+            for (const auto& data : field_datas) {
+                auto n = data->get_num_rows();
+                wrapper_->add_data<std::string>(
+                    static_cast<const std::string*>(data->Data()), n);
+            }
+            break;
+        }
+
+        case proto::schema::DataType::Array: {
+            switch (schema_.element_type()) {
+                case proto::schema::DataType::Bool: {
+                    build_index_for_array<bool>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int8: {
+                    build_index_for_array<int8_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int16: {
+                    build_index_for_array<int16_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int32: {
+                    build_index_for_array<int32_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int64: {
+                    build_index_for_array<int64_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Float: {
+                    build_index_for_array<float>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Double: {
+                    build_index_for_array<double>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::VarChar: {
+                    for (const auto& data : field_datas) {
+                        auto n = data->get_num_rows();
+                        auto array_column =
+                            static_cast<const Array*>(data->Data());
+                        for (int64_t i = 0; i < n; i++) {
+                            assert(
+                                array_column[i].get_element_type() ==
+                                static_cast<DataType>(schema_.element_type()));
+                            std::vector<std::string> output;
+                            for (int64_t j = 0; j < array_column[i].length();
+                                 j++) {
+                                output.push_back(
+                                    array_column[i]
+                                        .template get_data<std::string>(j));
+                            }
+                            wrapper_->template add_multi_data(output.data(),
+                                                              output.size());
```

Comment:
this code block can wrapper as specify template function for string

Reply:
OK.